angr/analyses/reaching_definitions/engine_vex.py

```diff
@@ -1090,12 +1090,15 @@ def _handle_function_cc(self, func_addr: Optional[MultiValues]):
                 elif isinstance(arg, SimStructArg):
                     min_stack_offset = None
                     for subargfield, subargloc in arg.locs.items():
-                        if not isinstance(subargloc, SimStackArg):
-                            raise TypeError(f"Unexpected: Field {subargfield} in {arg} is not a stack location.")
-                        if min_stack_offset is None:
-                            min_stack_offset = subargloc.stack_offset
-                        elif min_stack_offset > subargloc.stack_offset:
-                            min_stack_offset = subargloc.stack_offset
+                        if isinstance(subargloc, SimStackArg):
+                            if min_stack_offset is None:
+                                min_stack_offset = subargloc.stack_offset
+                            elif min_stack_offset > subargloc.stack_offset:
+                                min_stack_offset = subargloc.stack_offset
+                        elif isinstance(subargloc, SimRegArg):
+                            atom = Register(subargloc.reg_offset, subargloc.size)
+                            self.state.add_use(atom, code_loc)
+                            self._tag_definitions_of_atom(atom, func_addr_int)
```

Comment:
This looks like kind of a weird change. I'll need @ltfish to sign off on it.

Reply:
For documentation's sake, I ran into an issue with this aspect of engine_vex as the introduction of the O32 struct passing caused this area to fail since struct members can be passed into argument registers.

Reply:
LGTM